probe-rs-tools/src/bin/probe-rs/util/rtt.rs

```diff
@@ -308,14 +308,17 @@ impl RttActiveUpChannel {
                 };
 
                 // Format options:
-                // 1. Custom format for the channel
-                // 2. Default with optional timestamp and location
-                let format = if let Some(format) = channel_config.log_format.as_deref() {
-                    FormatterFormat::Custom(format)
-                } else {
-                    FormatterFormat::Default {
+                // 1. Oneline format with optional location
+                // 2. Custom format for the channel
+                // 3. Default with optional location
+                let format = match channel_config.log_format.as_deref() {
+                    Some("oneline") => FormatterFormat::OneLine {
```

Comment:
Can we introduce a proper enum for this instead of doing this string based? :) Maybe we can actually parse directly into the `FormatterFormat` enum?

Reply:
The RttChannelConfig seems to be defined here: https://github.com/probe-rs/probe-rs/blob/a121f2323a8c75a9be8a9cc5e413f59e8d57688d/probe-rs-tools/src/bin/probe-rs/util/rtt.rs#L54

It has `pub log_format: Option<String>`.

What do you propose instead? Arbitrary strings still need to be supported, but I'm hoping you might also have a pre-set list of "themes" or "styles" users can pick from without having to write out a whole format string.